### PR TITLE
Fix buildsystem

### PR DESCRIPTION
**Pull request by winterheart**
* Respect CC, CFLAGS, LDFLAGS
* Don't strip binaries on install (some Linux distributions requires non-stripped binaries for debug packages).

### CODE REVIEW

**Comment by WickedSmoke** [comment]
@winterheart, could you put the pkg-config changes into a separate commit?  I probably won't pull that, but we can get your other changes applied.

**Comment by winterheart** [comment]
Done.

File: Makefile [hunk 1]
```diff
@@ -6,7 +6,9 @@ ifeq ($(STATIC),1)
     STATIC_LIB=true
 endif
 
-CFLAGS=-O3 -DNDEBUG
+CC ?= cc
```
Comment (WickedSmoke) [inline Makefile]: GNU make defines CC, right?  Why was this added here but not in the Boron Makefile?

File: Makefile [hunk 2]
```diff
@@ -6,7 +6,9 @@ ifeq ($(STATIC),1)
...
+
+CFLAGS ?= -O3 -DNDEBUG
```
Comment (WickedSmoke) [inline Makefile]: In your Boron changes this is `CFLAGS += -O3 -DNDEBUG`.  Should we do that here as well?

File: Makefile (hunk 1 above)
Comment (winterheart) [reply Makefile]: Probably this redundant, just make sure that CC defined. Originally I made these changes for Gentoo Linux packaging faun and boron back in 2023 and don't quite remember my intentions :).

File: Makefile (hunk 2 above)
Comment (winterheart) [reply Makefile]: Probably yes. Will update PR.

**Comment by WickedSmoke** [comment]
The commit has been rebased on master.
